src/segment-loader.js

```diff
@@ -1048,10 +1081,8 @@ export default class SegmentLoader extends videojs.EventTarget {
         0, 0, 0, segmentInfo.mediaIndex + segmentInfo.playlist.mediaSequence
       ]);
 
-      simpleSegment.key = {
-        resolvedUri: segment.key.resolvedUri,
-        iv
-      };
+      simpleSegment.key = this.segmentKey(segment.key);
+      simpleSegment.key.iv = iv;
```

Comment:
Does this only work because the m3u8-parser repeats the last parsed `EXT-X-KEY` tag on every segment until it gets a different tag?

Reply:
correct